cmd/snap/error.go

```diff
@@ -190,3 +220,136 @@ If you understand and want to proceed repeat the command including --classic.
 
 	return msg, nil
 }
+
+func snapRevisionNotAvailableMessage(kind, snapName, action, arch, channel string, releases []interface{}) string {
+	req, err := snap.ParseChannel(channel, arch)
+	if err != nil {
+		msg := fmt.Sprintf(i18n.G("requested what looks like an invalid channel for snap %[1]q.\nPlease use 'snap info %[1]s' to list available releases."), snapName)
+		return msg
+	}
+	avail := make([]*snap.Channel, 0, len(releases))
+	for _, v := range releases {
+		rel, _ := v.(map[string]interface{})
+		relCh, _ := rel["channel"].(string)
+		relArch, _ := rel["architecture"].(string)
+		if relArch == "" {
+			logger.Debugf("internal error: revision-not-found store error carries a release with invalid/empty architecture: %v", v)
```

Comment:
why are we calling it a store error here, if it's from the daemon?
More concretely: either it's an internal error, or it's a store error :-)